src/rulesets/oas/__tests__/contact-properties.ts

```diff
@@ -1,13 +1,12 @@
-import { Spectral } from '../../../spectral';
-import { commonOasRules } from '../index';
-
-const ruleset = { rules: commonOasRules() };
+import { RuleType, Spectral } from '../../../spectral';
+import * as ruleset from '../ruleset.json';
 
 describe('contact-properties', () => {
   const s = new Spectral();
   s.addRules({
     'contact-properties': Object.assign(ruleset.rules['contact-properties'], {
       enabled: true,
+      type: RuleType[ruleset.rules['contact-properties'].type],
```

Comment:
What is the cause of us needing this? It seems like a pain in the backside to need to do this? Is this some sort of TypeScript to force the type to be a valid RuleType? Is there any other way of doing this?

Reply:
Yes, use a validation library (such as `io-ts`) or put `//ts-ignore` everywhere.

Reply:
Marking as resolved, was using `@ts-ignore` before and we agreed with @XVincentX this TS trick is better for now.